lib/matplotlib/ticker.py

```diff
@@ -931,51 +932,35 @@ def label_minor(self, labelOnlyBase):
         """
         self.labelOnlyBase = labelOnlyBase
 
+    def _numdec(self, vmin, vmax):
+        "Calculate number of decades"
+        vmin = math.log(vmin) / math.log(self._base)
+        vmax = math.log(vmax) / math.log(self._base)
+        return abs(vmax - vmin)
+
     def set_locs(self, locs=None):
         """
         Use axis view limits to control which ticks are labeled.
 
         The *locs* parameter is ignored in the present algorithm.
-
         """
         if np.isinf(self.minor_thresholds[0]):
             self._sublabels = None
             return
 
-        # Handle symlog case:
-        linthresh = self._linthresh
-        if linthresh is None:
-            try:
-                linthresh = self.axis.get_transform().linthresh
-            except AttributeError:
-                pass
-
         vmin, vmax = self.axis.get_view_interval()
         if vmin > vmax:
             vmin, vmax = vmax, vmin
 
-        if linthresh is None and vmin <= 0:
+        if vmin <= 0:
             # It's probably a colorbar with
             # a format kwarg setting a LogFormatter in the manner
             # that worked with 1.5.x, but that doesn't work now.
             self._sublabels = {1}  # label powers of base
             return
 
         b = self._base
-        if linthresh is not None:  # symlog
```

Comment:
This logic has moved into `_numdec` (either here or in the new `SymmetricalLogFormatter`.